docs/markdown/snippets/i18n_extract.md

```diff
@@ -0,0 +1,13 @@
+## i18n module find_xgettext
+
+There is a new `find_xgettext` function in `i18n` module that acts as a
+wrapper around `xgettext`. It allows to extract strings to translate from
+source files.
+
+This function is convenient, because:
+- It can find the sources files from a build target;
+- It will use an intermediate file when the number of source files is too
+  big to be handled directly from the command line;
+- It is able to get strings to translate from the dependencies of the given
+  targets.
```

Comment:
Ultimately the goal here is just to avoid the i18n.gettext() assumption that POTFILES is the source of truth for xgettext, right?

It feels like this function adds complexity which doesn't directly serve that. In particular, it seems odd to have a function which finds the *program*, and then have that yield an object with a single method. We could instead directly provide an `i18n.xgettext()` function with the same API contract which internally looks up xgettext, no?